File: src/main/java/mmo/core/network/Channel.java
package mmo.core.network;

import java.net.InetSocketAddress;

public interface Channel {
    /**
     * Get the channel id
     */
    Object id();

    /**
     * Write message to the channel
     *
     * @param message Message to send
     */
    void write(Object message);

    /**
     * Close the channel
     */
    void close();

    /**
     * Check if the channel is alive
     */
    boolean isAlive();

    /**
     * Get the client address
     */
    InetSocketAddress address();
}

File: src/main/java/mmo/core/network/Channel.kt
package mmo.core.network

import io.netty.channel.ChannelId
import java.net.InetSocketAddress

interface Channel {
    /**
     * Get the channel id
     */
    fun id(): ChannelId?

    /**
     * Write message to the channel
     *
     * @param message Message to send
     */
    fun write(message: Any?)

    /**
     * Close the channel
     */
    fun close()

    /**
     * Check if the channel is alive
     */
    val isAlive: Boolean

    /**
     * Get the client address
     */
    fun address(): InetSocketAddress
}